src/nifake/unit_tests/test_grpc.py

```diff
@@ -161,8 +164,73 @@ def _assert_call(self, function_name, request_object):
         getattr(self.patched_grpc_stub, function_name).assert_called_once_with(request_object)
         return getattr(self.patched_grpc_types, function_name + 'Request')
 
+    # gRPC errors
+
+    def test_server_unavailable(self):
+        library_func = 'InitWithOptions'
+        grpc_error = grpc.StatusCode.UNAVAILABLE
+        expected_error_message = 'Failed to connect to server'
+        self._set_side_effect(library_func, side_effect=MyRpcError(None, "", grpc_error=grpc_error))
+        grpc_options = nifake.GrpcSessionOptions(object(), "", nifake.SessionInitializationBehavior.AUTO)
+        interpreter = nifake._grpc_stub_interpreter.GrpcStubInterpreter(grpc_options)
+        try:
+            interpreter.init_with_options('dev1', False, False, '')
+            assert False
+        except nifake.Error as e:
+            assert e.rpc_code == grpc_error
+            assert e.description == expected_error_message
+            assert str(e) == f'StatusCode.UNAVAILABLE: {expected_error_message}'
+
+    def test_function_not_implemented(self):
+        library_func = 'PoorlyNamedSimpleFunction'
+        grpc_error = grpc.StatusCode.UNIMPLEMENTED
+        expected_error_message_intro = 'This operation is not supported'
+        self._set_side_effect(library_func, side_effect=MyRpcError(None, "", grpc_error=grpc_error))
+        interpreter = self._get_initialized_library_interpreter()
```

Comment:
This being in the gRPC section, it seems odd to use the library interpreter.

Reply:
Already being changed [here](https://github.com/ni/nimi-python/pull/1840/files#diff-899f0ea835ab662dd4436c04e6f4de2ccc24050445a2871ef3bda7256bcee85b). Whichever of these gets merged first, the other will have to merge and update, and then it'll be correct.